lib/puppet/provider/oneview_storage_pool/c7000.rb

```diff
@@ -60,4 +67,10 @@ def setup_for_recreation
     storage_pool.first.remove
     true
   end
+
+  def set_storage_system
+    return unless @data['storageSystemUri'].empty?
+    storage_system_class = OneviewSDK.resource_named('StorageSystem', @client.api_version)
+    @data['storageSystemUri'] = storage_system_class.get_all(@client).first['uri']
```

Comment:
we can have multiple storage systems added to oneview. So please let the user to provide the name or ip of storage system and you use "find_by" method using that name.
```
OneviewSDK.resource_named('StoragePool', api_version).find_by(@client, name: @data['storageSystem']).first['uri']
```

Reply:
Done.